_posts/2016-03-25-hello-squabble.md

```diff
@@ -0,0 +1,49 @@
+
+### Built on Lumen
+
+Looking around the open source landscape for the right tool to build this app, I came across [Lumen](https://lumen.laravel.com/), a slimmed down version of the well-known [Laravel](https://www.laravel.com) project. Lumen was very simple to get up and running with thanks to the [clear documentation examples](https://lumen.laravel.com/docs/5.2).
+
+The entry point of our application are some [routing rules](https://github.com/savaslabs/squabble/blob/master/app/Http/routes.php) that determine what data to give back to the user when they visit a particular API endpoint.
+
+From there, the requests are processed in the [CommentController](https://github.com/savaslabs/squabble/blob/master/app/Http/Controllers/CommentController.php) — depending on the request, we can create a comment, retrieve a comment, delete it, or get a quick list of all posts and the number of comments per page (we use this information on our home page and on the [News](/news) page to show the comment count per article).
+
+The application runs on one of our cloud servers on Linode, and uses a SQLite backend. Because the Lumen framework is very lean, and the application does relatively little, it's very fast — the `api/comments/count` call takes ~84 ms while loading all comments (`api/comments`) takes about 125 ms.
+
+### Tying it all together
+
+Now that we've set up our backend application for managing comments, we need a way to interact with it. On this site, we wrote some JavaScript to interact with the API. You can look at our Jekyll site's [`main.js`](https://github.com/savaslabs/savaslabs.github.io/blob/master/assets/js/main.js) for the gory details, but basically we make use of the jQuery `ajax()` method to `GET` and `POST` data from and to our commenting API.
+
+## Conclusion
+
+And now you know how and why the "Comments" section on this post appears as it does. In conclusion, I should mention that the major downside of building your own comment hosting application is that you have to build your own comment hosting application. In other words, nothing comes for free — you have to build every feature. That's why, for example, we don't (yet) have nicely formatted comments (they are plain text only), and users don't get notifications when someone replies to their comment, there's no comment threading, etc. On the other hand, we don't really need those features at the moment, and it's been a good learning experience for our team to develop this application over the last year. Our [website](https://github.com/savaslabs/savaslabs.github.io) and [comment application](https://github.com/savaslabs/squabble) code is all open source, so have a look and feel free to use it for your own projects.
+
+If you've made it all the way to the end of this post, then go on, and leave a comment!
+
+[^1]: [Nearly](https://twitter.com/githubstatus/status/711965206029725697).
```

Comment:
I think labelling it "footnotes" is a good idea. I went to far as to style the div with class `footnotes`, but we could think about adding this to the front matter so the template can build it out in a consistent way. Then again, dealing with lots of front matter can get annoying at some point.